Error: ENOENT: no such file or directory, scandir '/workspaces/shipment-tracker/frontend/build'

When I follow the Dev Setup instructions I get this error:

```
Error: ENOENT: no such file or directory, scandir '/workspaces/shipment-tracker/frontend/build'
    at Object.readdirSync (fs.js:1021:3)
    at listFiles (/workspaces/shipment-tracker/src/getAllFilesSync.ts:6:32)
    at Object.getAllFilesSync [as default] (/workspaces/shipment-tracker/src/getAllFilesSync.ts:21:10)
    at Object.<anonymous> (/workspaces/shipment-tracker/src/server.ts:41:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Module._compile (/workspaces/shipment-tracker/node_modules/source-map-support/source-map-support.js:547:25)
    at Module.m._compile (/tmp/ts-node-dev-hook-9590723593235797.js:69:33)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at require.extensions.<computed> (/tmp/ts-node-dev-hook-9590723593235797.js:71:20)
    at Object.nodeDevHook [as .ts] (/workspaces/shipment-tracker/node_modules/ts-node-dev/lib/hook.js:63:13)
[ERROR] 10:50:20 Error: ENOENT: no such file or directory, scandir '/workspaces/shipment-tracker/frontend/build'
```

<details>
 <summary>Full log</summary>

    👋 Welcome to Codespaces! You are on our default image. It includes runtimes and tools for: 
    - Python 
    - Node.js
    - Docker
    - And more! To see the full list, go here: https://aka.ms/ghcs-default-image

    🔍 To explore VS Code to its fullest, search using the Command Palette (Cmd/Ctrl + Shift + P or F1).

    📝 Edit away, run your app as usual, and we'll automatically make it available for you to access.

    codespace ➜ /workspaces/shipment-tracker (saga) $ script/dev_setup
    Setting up dev environment

    + Detected node v14, using that...
    + yarn already installed... skipping.
    yarn install v1.22.10
    [1/5] Validating package.json...
    [2/5] Resolving packages...
    [3/5] Fetching packages...
    info fsevents@2.3.2: The platform "linux" is incompatible with this module.
    info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
    [4/5] Linking dependencies...
    warning "graphql-tools > @graphql-tools/links@7.0.5" has unmet peer dependency "@apollo/client@~3.2.5 || ~3.3.0".
    warning " > react-transition-group@4.4.1" has unmet peer dependency "react@>=16.6.0".
    warning " > react-transition-group@4.4.1" has unmet peer dependency "react-dom@>=16.6.0".
    warning " > @graphql-codegen/typescript-react-apollo@2.2.2" has unmet peer dependency "graphql-tag@^2.0.0".
    [5/5] Building fresh packages...
    $ husky install
    husky - Git hooks installed
    Done in 7.58s.
    + Installing postgresql...
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    The following additional packages will be installed:
    cron libsensors-config libsensors5 libxslt1.1 logrotate postgresql-12 postgresql-client-12 postgresql-client-common postgresql-common ssl-cert sysstat
    Suggested packages:
    anacron checksecurity default-mta | mail-transport-agent lm-sensors bsd-mailx | mailx postgresql-doc postgresql-doc-12 libjson-perl openssl-blacklist isag
    The following NEW packages will be installed:
    cron libsensors-config libsensors5 libxslt1.1 logrotate postgresql postgresql-12 postgresql-client-12 postgresql-client-common postgresql-common ssl-cert sysstat
    0 upgraded, 12 newly installed, 0 to remove and 4 not upgraded.
    Need to get 15.5 MB of archives.
    After this operation, 48.4 MB of additional disk space will be used.
    Get:1 http://archive.ubuntu.com/ubuntu focal/main amd64 cron amd64 3.0pl1-136ubuntu1 [71.5 kB]
    Get:2 http://archive.ubuntu.com/ubuntu focal/main amd64 logrotate amd64 3.14.0-4ubuntu3 [44.5 kB]
    Get:3 http://archive.ubuntu.com/ubuntu focal/main amd64 libsensors-config all 1:3.6.0-2ubuntu1 [6,092 B]
    Get:4 http://archive.ubuntu.com/ubuntu focal/main amd64 libsensors5 amd64 1:3.6.0-2ubuntu1 [27.4 kB]
    Get:5 http://archive.ubuntu.com/ubuntu focal/main amd64 libxslt1.1 amd64 1.1.34-4 [152 kB]
    Get:6 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postgresql-client-common all 214ubuntu0.1 [28.2 kB]
    Get:7 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postgresql-client-12 amd64 12.6-0ubuntu0.20.04.1 [1,045 kB]
    Get:8 http://archive.ubuntu.com/ubuntu focal/main amd64 ssl-cert all 1.0.39 [17.0 kB]
    Get:9 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postgresql-common all 214ubuntu0.1 [169 kB]
    Get:10 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postgresql-12 amd64 12.6-0ubuntu0.20.04.1 [13.5 MB]
    Get:11 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postgresql all 12+214ubuntu0.1 [3,924 B]
    Get:12 http://archive.ubuntu.com/ubuntu focal/main amd64 sysstat amd64 12.2.0-2 [453 kB]
    Fetched 15.5 MB in 0s (41.2 MB/s)
    Preconfiguring packages ...
    Selecting previously unselected package cron.
    (Reading database ... 117231 files and directories currently installed.)
    Preparing to unpack .../00-cron_3.0pl1-136ubuntu1_amd64.deb ...
    Unpacking cron (3.0pl1-136ubuntu1) ...
    Selecting previously unselected package logrotate.
    Preparing to unpack .../01-logrotate_3.14.0-4ubuntu3_amd64.deb ...
    Unpacking logrotate (3.14.0-4ubuntu3) ...
    Selecting previously unselected package libsensors-config.
    Preparing to unpack .../02-libsensors-config_1%3a3.6.0-2ubuntu1_all.deb ...
    Unpacking libsensors-config (1:3.6.0-2ubuntu1) ...
    Selecting previously unselected package libsensors5:amd64.
    Preparing to unpack .../03-libsensors5_1%3a3.6.0-2ubuntu1_amd64.deb ...
    Unpacking libsensors5:amd64 (1:3.6.0-2ubuntu1) ...
    Selecting previously unselected package libxslt1.1:amd64.
    Preparing to unpack .../04-libxslt1.1_1.1.34-4_amd64.deb ...
    Unpacking libxslt1.1:amd64 (1.1.34-4) ...
    Selecting previously unselected package postgresql-client-common.
    Preparing to unpack .../05-postgresql-client-common_214ubuntu0.1_all.deb ...
    Unpacking postgresql-client-common (214ubuntu0.1) ...
    Selecting previously unselected package postgresql-client-12.
    Preparing to unpack .../06-postgresql-client-12_12.6-0ubuntu0.20.04.1_amd64.deb ...
    Unpacking postgresql-client-12 (12.6-0ubuntu0.20.04.1) ...
    Selecting previously unselected package ssl-cert.
    Preparing to unpack .../07-ssl-cert_1.0.39_all.deb ...
    Unpacking ssl-cert (1.0.39) ...
    Selecting previously unselected package postgresql-common.
    Preparing to unpack .../08-postgresql-common_214ubuntu0.1_all.deb ...
    Adding 'diversion of /usr/bin/pg_config to /usr/bin/pg_config.libpq-dev by postgresql-common'
    Unpacking postgresql-common (214ubuntu0.1) ...
    Selecting previously unselected package postgresql-12.
    Preparing to unpack .../09-postgresql-12_12.6-0ubuntu0.20.04.1_amd64.deb ...
    Unpacking postgresql-12 (12.6-0ubuntu0.20.04.1) ...
    Selecting previously unselected package postgresql.
    Preparing to unpack .../10-postgresql_12+214ubuntu0.1_all.deb ...
    Unpacking postgresql (12+214ubuntu0.1) ...
    Selecting previously unselected package sysstat.
    Preparing to unpack .../11-sysstat_12.2.0-2_amd64.deb ...
    Unpacking sysstat (12.2.0-2) ...
    Setting up logrotate (3.14.0-4ubuntu3) ...
    Created symlink /etc/systemd/system/timers.target.wants/logrotate.timer → /lib/systemd/system/logrotate.timer.
    Setting up postgresql-client-common (214ubuntu0.1) ...
    Setting up cron (3.0pl1-136ubuntu1) ...
    Adding group `crontab' (GID 108) ...
    Done.
    invoke-rc.d: could not determine current runlevel
    invoke-rc.d: policy-rc.d denied execution of start.
    Created symlink /etc/systemd/system/multi-user.target.wants/cron.service → /lib/systemd/system/cron.service.
    Setting up libsensors-config (1:3.6.0-2ubuntu1) ...
    Setting up postgresql-client-12 (12.6-0ubuntu0.20.04.1) ...
    update-alternatives: using /usr/share/postgresql/12/man/man1/psql.1.gz to provide /usr/share/man/man1/psql.1.gz (psql.1.gz) in auto mode
    Setting up ssl-cert (1.0.39) ...
    Setting up postgresql-common (214ubuntu0.1) ...
    Adding user postgres to group ssl-cert

    Creating config file /etc/postgresql-common/createcluster.conf with new version
    Building PostgreSQL dictionaries from installed myspell/hunspell packages...
    Removing obsolete dictionary files:
    Created symlink /etc/systemd/system/multi-user.target.wants/postgresql.service → /lib/systemd/system/postgresql.service.
    invoke-rc.d: could not determine current runlevel
    invoke-rc.d: policy-rc.d denied execution of start.
    Setting up libsensors5:amd64 (1:3.6.0-2ubuntu1) ...
    Setting up libxslt1.1:amd64 (1.1.34-4) ...
    Setting up postgresql-12 (12.6-0ubuntu0.20.04.1) ...
    Creating new PostgreSQL cluster 12/main ...
    /usr/lib/postgresql/12/bin/initdb -D /var/lib/postgresql/12/main --auth-local peer --auth-host md5
    The files belonging to this database system will be owned by user "postgres".
    This user must also own the server process.

    The database cluster will be initialized with locale "C.UTF-8".
    The default database encoding has accordingly been set to "UTF8".
    The default text search configuration will be set to "english".

    Data page checksums are disabled.

    fixing permissions on existing directory /var/lib/postgresql/12/main ... ok
    creating subdirectories ... ok
    selecting dynamic shared memory implementation ... posix
    selecting default max_connections ... 100
    selecting default shared_buffers ... 128MB
    selecting default time zone ... Etc/UTC
    creating configuration files ... ok
    running bootstrap script ... ok
    performing post-bootstrap initialization ... ok
    syncing data to disk ... ok

    Success. You can now start the database server using:

        pg_ctlcluster 12 main start

    Ver Cluster Port Status Owner    Data directory              Log file
    12  main    5432 down   postgres /var/lib/postgresql/12/main /var/log/postgresql/postgresql-12-main.log
    update-alternatives: using /usr/share/postgresql/12/man/man1/postmaster.1.gz to provide /usr/share/man/man1/postmaster.1.gz (postmaster.1.gz) in auto mode
    invoke-rc.d: could not determine current runlevel
    invoke-rc.d: policy-rc.d denied execution of start.
    Setting up sysstat (12.2.0-2) ...

    Creating config file /etc/default/sysstat with new version
    update-alternatives: using /usr/bin/sar.sysstat to provide /usr/bin/sar (sar) in auto mode
    Created symlink /etc/systemd/system/multi-user.target.wants/sysstat.service → /lib/systemd/system/sysstat.service.
    Setting up postgresql (12+214ubuntu0.1) ...
    Processing triggers for systemd (245.4-4ubuntu3.6) ...
    Processing triggers for man-db (2.9.1-1) ...
    Processing triggers for libc-bin (2.31-0ubuntu9.2) ...
    + Installing postgresql-client...
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    The following NEW packages will be installed:
    postgresql-client
    0 upgraded, 1 newly installed, 0 to remove and 4 not upgraded.
    Need to get 3,940 B of archives.
    After this operation, 67.6 kB of additional disk space will be used.
    Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postgresql-client all 12+214ubuntu0.1 [3,940 B]
    Fetched 3,940 B in 0s (146 kB/s)              
    Selecting previously unselected package postgresql-client.
    (Reading database ... 119132 files and directories currently installed.)
    Preparing to unpack .../postgresql-client_12+214ubuntu0.1_all.deb ...
    Unpacking postgresql-client (12+214ubuntu0.1) ...
    Setting up postgresql-client (12+214ubuntu0.1) ...
    + Starting postgresql server...
    postgresql-13.2: unrecognized service
    codespace ➜ /workspaces/shipment-tracker (saga) $ yarn dev
    yarn run v1.22.10
    $ yarn codegen
    $ graphql-codegen
    ✔ Parse configuration
    ✔ Generate outputs
    $ npm-run-all --parallel dev:server codegen:watch
    $ ts-node-dev --no-notify --respawn --transpile-only src/server
    $ graphql-codegen --watch
    [INFO] 10:50:18 ts-node-dev ver. 1.1.1 (using ts-node ver. 9.1.1, typescript ver. 4.1.5)
    ✔ Parse configuration
    ✔ Generate outputs
    ℹ Watching for changes...
    Error: ENOENT: no such file or directory, scandir '/workspaces/shipment-tracker/frontend/build'
        at Object.readdirSync (fs.js:1021:3)
        at listFiles (/workspaces/shipment-tracker/src/getAllFilesSync.ts:6:32)
        at Object.getAllFilesSync [as default] (/workspaces/shipment-tracker/src/getAllFilesSync.ts:21:10)
        at Object.<anonymous> (/workspaces/shipment-tracker/src/server.ts:41:16)
        at Module._compile (internal/modules/cjs/loader.js:1063:30)
        at Module._compile (/workspaces/shipment-tracker/node_modules/source-map-support/source-map-support.js:547:25)
        at Module.m._compile (/tmp/ts-node-dev-hook-9590723593235797.js:69:33)
        at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
        at require.extensions.<computed> (/tmp/ts-node-dev-hook-9590723593235797.js:71:20)
        at Object.nodeDevHook [as .ts] (/workspaces/shipment-tracker/node_modules/ts-node-dev/lib/hook.js:63:13)
    [ERROR] 10:50:20 Error: ENOENT: no such file or directory, scandir '/workspaces/shipment-tracker/frontend/build'

</details>

STR:

1. Open project in CodeSpaces
2. run `script/dev_setup`
3. run `yarn dev`

--- a/script/dev_setup
+++ b/script/dev_setup
@@ -175,6 +175,7 @@ fi
 copy_env_file
 
 # Build frontend
+yarn run codegen
 pushd frontend && yarn install && yarn run build:styles && yarn build && popd
 
 echo

